pkg/config/config.go

```diff
@@ -61,91 +61,59 @@ type Config struct {
 	EvalContext *hcl.EvalContext
 }
 
-func loadLokocfgPaths(configPath string) ([]string, error) {
-	isDir, err := util.PathIsDir(configPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to stat config path %q: %w", configPath, err)
-	}
-	var lokocfgPaths []string
-	if isDir {
-		globPattern := filepath.Join(configPath, "*.lokocfg")
-		configFiles, err := filepath.Glob(globPattern)
-		if err != nil {
-			return nil, fmt.Errorf("bad filepath glob pattern %q: %w", globPattern, err)
+// Component finds a component named name and returns its configuration as a pointer to an
+// hcl.Body. If the component isn't found, nil is returned.
+func (c *Config) Component(name string) *hcl.Body {
+	for _, component := range c.RootConfig.Components {
+		if name == component.Name {
+			return &component.Config
 		}
-		lokocfgPaths = append(lokocfgPaths, configFiles...)
-	} else {
-		lokocfgPaths = append(lokocfgPaths, configPath)
 	}
-	return lokocfgPaths, nil
+
+	return nil
 }
 
-func LoadConfig(lokocfgPath, lokocfgVarsPath string) (*Config, hcl.Diagnostics) {
-	lokocfgPaths, err := loadLokocfgPaths(lokocfgPath)
+// ReadHCL reads all HCL files at path and one HCL variables file at varPath, constructs a Config
+// and returns a pointer to it.
+func ReadHCL(path, varPath string) (*Config, hcl.Diagnostics) {
+	names, err := lokocfgFilenames(path)
 	if err != nil {
-		return nil, hcl.Diagnostics{
-			&hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  err.Error(),
-			},
-		}
+		return nil, diag(fmt.Sprintf("getting lokocfg paths: %v", err))
 	}
 
-	hclParser := hclparse.NewParser()
-
-	var hclFiles []*hcl.File
-	for _, f := range lokocfgPaths {
-		hclFile, diags := hclParser.ParseHCLFile(f)
-		if len(diags) > 0 {
-			return nil, diags
-		}
-		hclFiles = append(hclFiles, hclFile)
+	files, err := readFiles(names)
+	if err != nil {
+		return nil, diag(err.Error())
 	}
 
-	configBody := hcl.MergeFiles(hclFiles)
+	body, diags := parseHCLFiles(files)
+	if diags.HasErrors() {
+		return nil, diags
+	}
```

Comment:
To avoid such lengthy functions and to name the blocks of code, I think we could put this three actions into separate function, but I'm not sure how to phrase the name of this function. However, after looking at this code (and code after it), it's clear to me, that this part reads the configuration files (and not values files) and returns them in a form of HCL body.

Reply:
I don't think `ReadHCL()` is so long at the moment, and breaking it down further would start to be on the "too much indirection" side. *However*, looking at the function again I'm not happy with the name `ReadHCL` since this function does more than just read HCL files. Frankly I think `config.New()` was the best option from the discussions so far. Do you have an alternative which doesn't include the word "load"? If not, I'd just go with `config.New()`.

Reply:
>I don't think ReadHCL() is so long at the moment, and breaking it down further would start to be on the "too much indirection" side.

Given that this is a exported function, so we should be able to cover it fully with tests, currently has 9 different paths, which IMO is to high cognitive complexity. In https://github.com/kinvolk/lokomotive/commit/8c76770cae2af3ed53b9733af3967736ac11fe72 I broke it down into nice smaller functions, so we have only 4 paths, which is very reasonable and easy to understand. This is also what tools like https://codeclimate.com/ encourages to do.

I think with the right naming you don't create the indirection, quite the opposite. You hide unrelated details, which makes things simpler and allows understanding only the paths of code which are interesting for the reader. With long function, reader must go trough all of it and understand all the statements to analyze the function. 

>However, looking at the function again I'm not happy with the name ReadHCL since this function does more than just read HCL files. Frankly I think config.New() was the best option from the discussions so far. Do you have an alternative which doesn't include the word "load"? If not, I'd just go with config.New().

`FromHCLFiles()` maybe? Or `ReadHCLFiles()` ? It gives you configuration from HCL files, so I think the name is appropriate. `config.New()` would be good IMO if we could define the loader type as an argument. Then the name shouldn't imply from where or how do we load the configuration. However, if we take HCL files paths as an argument, name should reflect that too in my opinion.

Reply:
@johananl I'm currently looking into untangling and isolating code from `cli/cmd` package, as part of #312 and at the moment I find it confusing, that we call `config.LoadConfig()`, but it doesn't really load entire configuration. It's even hard to define of what exactly this code is doing. Maybe that will help a bit with naming and documentation here.

Reply:
This is the definition of existing `LoadConfig` function I came up with:
```go
// LoadConfig reads and parses top-level HCL configuration from given lokocfgPath and
// evaluates the variables from lokocfgVarsPath into HCL context, so it can be used
// for futher configuration parsing.
```

Which leads to the conclusion, that what is being returned is not the configuration (as "user configuration"), but rather a facility, which allows you to further read the configuration.

The `lokomotive/pkg/config.Config` struct does not ship proper information about itself. It's not "a Lokomotive configuration". It does not represent cluster + components configuration (which we didn't name yet).